### ISSUE TITLE
JIT tracing bug with STV in conjunction with non proportional method (only PHP 8.0 with JIT tracing mode enabled)

### ISSUE THREAD

**Issue by julien-boudry** [julien-boudry/Condorcet#61]
| Q                   | A
| --------------------| ---------------
| Type                | Bug / Support / Suggestion / Other
| Condorcet version   | 3.1.0 && 3.2.0
| PHP version         | 8.0.10 (NTS, x64, Linux + Windows) && PHP 8.1-RC6
| Installation Method |  __CondorcetAutoload.php

[See this failing code](https://github.com/julien-boudry/Condorcet/blob/master/Dev/bugs/JitBug.php)

_just git clone from actual master branch for 3.2 branch with PHP 8.1 ; branch dev-3.1 for 3.1 with PHP 8.0_
```
Linux:
php -dzend_extension=opcache -dopcache.enable_cli=1 -dopcache.jit_buffer_size=100M -dopcache.jit=tracing Dev/bugs/JitBug.php

Windows:
php -d zend_extension=opcache -d opcache.enable_cli=1 -d opcache.jit_buffer_size=100M -d opcache.jit=tracing Dev/bugs/JitBug.php
```

Unable to reproduce in theoric PHP (after many trials), it's a JIT optimization bug that only happens in Condorcet when using STV in conjunction with a non-proportional method.

* The bug only with jit "tracing" mode (working with jit "function" mode) (test on PHP 8.0.10 NTS on x64)
* Working with or without opcache, the bug only if using jit tracing mode (which is the default for JIT)
* It's working fine with PHP 8.1-RC1... and it's buggy with RC6
* No bug if you remove named parameter from the call (`new Result ([...]`)

**The bug:**

```
PHP Fatal error:  Uncaught TypeError: CondorcetPHP\Condorcet\Result::__construct(): Argument #6 ($seats) must be of type ?int, CondorcetPHP\Condorcet\Election given, called in C:\dev_scripts\Condorcet\lib\Algo\Method.php on line 84 and defined in C:\dev_scripts\Condorcet\lib\Result.php:89
Stack trace:
#0 C:\dev_scripts\Condorcet\lib\Algo\Method.php(84): CondorcetPHP\Condorcet\Result->__construct()
#1 C:\dev_scripts\Condorcet\lib\Algo\Methods\STV\SingleTransferableVote.php(91): CondorcetPHP\Condorcet\Algo\Method->createResult()
#2 C:\dev_scripts\Condorcet\lib\Algo\Method.php(63): CondorcetPHP\Condorcet\Algo\Methods\STV\SingleTransferableVote->compute()
#3 C:\dev_scripts\Condorcet\lib\ElectionProcess\ResultsProcess.php(80): CondorcetPHP\Condorcet\Algo\Method->getResult()
#4 C:\dev_scripts\Condorcet\Dev\bugs\JitBug.php(37): CondorcetPHP\Condorcet\Election->getResult()
#5 {main}
```

It looks impossible if you have a look on `Condorcet\lib\Algo\Method.php(84): CondorcetPHP\Condorcet\Result->__construct()` and `Election->getNumberOfSeats()` which is typed to integer.

**Comment by julien-boudry** [comment]
https://bugs.php.net/bug.php?id=81432

**Comment by julien-boudry** [comment]
Another bug with same JitBug.php since commit #859c71c with PGP 8.1-RC2
https://bugs.php.net/bug.php?id=81453

**Comment by julien-boudry** [comment]
Fixed in last PHP release (8.0 and 8.1)
